src/NexusMods.App.UI/Pages/LibraryPage/LocalFileLibraryItemModel.cs

```diff
@@ -0,0 +1,101 @@
+using DynamicData;
+using NexusMods.Abstractions.Library.Models;
+using NexusMods.Abstractions.Loadouts;
+using NexusMods.App.UI.Controls;
+using NexusMods.App.UI.Extensions;
+using NexusMods.MnemonicDB.Abstractions;
+using NexusMods.Paths;
+using ObservableCollections;
+using R3;
+
+namespace NexusMods.App.UI.Pages.LibraryPage;
+
+public class LocalFileLibraryItemModel : TreeDataGridItemModel<ILibraryItemModel, EntityId>,
+    ILibraryItemWithName,
+    ILibraryItemWithSize,
+    ILibraryItemWithDates,
+    ILibraryItemWithInstallAction,
+    IHasLinkedLoadoutItems,
+    IIsChildLibraryItemModel
+{
+    public LocalFileLibraryItemModel(LocalFile.ReadOnly localFile)
+    {
+        LibraryItemId = localFile.Id;
+
+        FormattedSize = ItemSize.ToFormattedProperty();
+        FormattedDownloadedDate = DownloadedDate.ToFormattedProperty();
+        FormattedInstalledDate = InstalledDate.ToFormattedProperty();
+        InstallItemCommand = ILibraryItemWithInstallAction.CreateCommand(this);
+
+        // ReSharper disable once NotDisposedResource
+        var datesDisposable = ILibraryItemWithDates.SetupDates(this);
+
+        var linkedLoadoutItemsDisposable = new SerialDisposable();
+
+        // ReSharper disable once NotDisposedResource
+        var modelActivationDisposable = this.WhenActivated(linkedLoadoutItemsDisposable, static (self, linkedLoadoutItemsDisposable, disposables) =>
+        {
+            // ReSharper disable once NotDisposedResource
+            IHasLinkedLoadoutItems.SetupLinkedLoadoutItems(self, linkedLoadoutItemsDisposable).AddTo(disposables);
+        });
+
+        _modelDisposable = Disposable.Combine(
+            datesDisposable,
+            linkedLoadoutItemsDisposable,
+            modelActivationDisposable,
+            Name,
+            ItemSize,
+            FormattedSize,
+            DownloadedDate,
+            FormattedDownloadedDate,
+            InstalledDate,
+            FormattedInstalledDate,
+            InstallItemCommand,
+            IsInstalled,
+            InstallButtonText
+        );
+    }
```

Comment:
I hate we have to do this. 
It seems way to easy to just forget something, or not add it here when something new is added.

Reply:
Not much you can do about that.

Reply:
:(

Reply:
:(